torch/package/importer.py

```diff
@@ -141,8 +142,23 @@ def whichmodule(self, obj: Any, name: str) -> str:
         module_name = getattr(obj, "__module__", None)
         if module_name is not None:
             return module_name
-        else:
-            return "__main__"
+
+        # Protect the iteration by using a list copy of self.modules against dynamic
+        # modules that trigger imports of other modules upon calls to getattr.
+        for module_name, module in self.modules.copy().items():
+            if (
+                module_name == "__main__"
+                or module_name == "__mp_main__"  # bpo-42406
+                or module is None
+            ):
+                continue
+            try:
+                if _getattribute(module, name)[0] is obj:
```

Comment:
Why `_getattribute` from `pickle` instead of regular `getattr`? `help(pickle._getattribute)` is empty. 😕

Reply:
it will accordion out `name` which may be a qualified path (`foo.bar.baz`). No big difference otherwise. But this is just copypasta from the pickle code